frappe/email/doctype/notification/notification.py

```diff
@@ -167,7 +167,10 @@ def create_system_notification(self, doc, context):
 			subject = frappe.render_template(self.subject, context)
 
 		attachments = self.get_attachment(doc)
+
 		recipients, cc, bcc = self.get_list_of_recipients(doc, context)
+		if not (recipients or cc or bcc):
```

Comment:
```suggestion
		if not (recipients and cc and bcc):
```

Reply:
@surajshetty3416 we need to return only when all three are `None`

Reply:
@prssanna The suggestion was based on: if anyone (recipients, cc or bcc) is `None` then the next line which is `users = recipients + cc + bcc` will fail.

If the value returned from `get_list_of_recipients` will be specific, we can simplify the check with `if recipients is None:` we do not even need to check others.

Reply:
This condition is meant to just specifically handle this case:
https://github.com/frappe/frappe/blob/5bf20116f08654fb179b0c609db83957713bb96f/frappe/email/doctype/notification/notification.py#L283-L285

Otherwise, all three returned values will be a list.